workbench-core/base/src/utilities/validatorHelper.ts

```diff
@@ -74,9 +78,14 @@ z.ZodString.prototype.etcId = function (): ZodString {
 };
 
 z.ZodString.prototype.nonEmpty = function (): ZodString {
+  // field should be nonEmpty but is optional
```

Comment:
It isn't clear that `nonEmpty` means optional and not empty. I'd expect nonEmpty means required. Are you sure the comment is accurate? We may want to change the name / functionality to have optional but non-empty if present to be something like `z.string().nonEmpty().optional`

Reply:
rename the method to optionalNonEmpty() to include optional() check